File: backend/src/main/java/med/doctor_connect/service/PrescriptionService.java
package med.doctor_connect.service;

import med.doctor_connect.dto.CreatePrescriptionRequest;
import med.doctor_connect.dto.PrescriptionDto;
import org.springframework.data.domain.Page;

import java.util.UUID;

public interface PrescriptionService {

    PrescriptionDto createPrescription(UUID appointmentId, CreatePrescriptionRequest request);

    PrescriptionDto getPrescriptionById(UUID id);

    Page<PrescriptionDto> getPatientPrescriptions(UUID patientId, int page, int limit);
}

File: frontend/doctor-connect-ui/src/app/services/prescription.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Prescription, CreatePrescriptionRequest, PaginatedResponse } from '../models';

@Injectable({
  providedIn: 'root'
})
export class PrescriptionService {
  private readonly API_URL = 'http://localhost:8000/api/v1';

  constructor(private http: HttpClient) {}

  createPrescription(appointmentId: string, request: CreatePrescriptionRequest): Observable<Prescription> {
    return this.http.post<Prescription>(`${this.API_URL}/appointments/${appointmentId}/prescriptions`, request);
  }

  getPrescriptionById(prescriptionId: string): Observable<Prescription> {
    return this.http.get<Prescription>(`${this.API_URL}/prescriptions/${prescriptionId}`);
  }

  getPatientPrescriptions(patientId: string, page: number = 0, limit: number = 20): Observable<PaginatedResponse<Prescription>> {
    const params = new HttpParams()
      .set('page', page.toString())
      .set('limit', limit.toString());

    return this.http.get<PaginatedResponse<Prescription>>(`${this.API_URL}/patients/${patientId}/prescriptions`, { params });
  }
}
